src/Microsoft.AspNetCore.Razor.Evolution/Legacy/TagHelperParseTreeRewriter.cs

```diff
@@ -471,7 +471,7 @@ private bool HasAllowedChildren()
                 return false;
             }
 
-            return _currentTagHelperTracker.AllowedChildren != null;
+            return _currentTagHelperTracker.AllowedChildren != null && _currentTagHelperTracker.AllowedChildren.Count() > 0;
```

Comment:
Why is this `.Count()`? Aren't all of these things `IReadOnlyList<>`? If they aren't then they should be.